coderd/workspacebuilds.go

```diff
@@ -404,6 +404,16 @@ func (api *API) postWorkspaceBuilds(rw http.ResponseWriter, r *http.Request) {
 			ctx,
 			tx,
 			func(action policy.Action, object rbac.Objecter) bool {
+				// Special handling for prebuilt workspace deletion
+				if object.RBACObject().Type == rbac.ResourceWorkspace.Type && action == policy.ActionDelete {
+					if workspaceObj, ok := object.(database.Workspace); ok {
+						// Try prebuilt-specific authorization first
+						if auth := api.Authorize(r, action, workspaceObj.AsPrebuild()); auth {
+							return auth
+						}
+					}
```

Comment:
What you're doing here is checking prebuild delete auth if a user is trying to delete a workspae regardless of whether the workspace is a prebuild. I think we only want to do this if the workspace is a prebuild workspace, and then strictly return the auth for that object and otherwise fall back to regular workspace auth.

What do you think about this instead? 

```suggestion
					if workspaceObj, ok := object.(database.Workspace); ok && workspaceObj.isPrebuild() {
						// Try prebuilt-specific authorization first
						return api.Authorize(r, action, workspaceObj.AsPrebuild())
					}
```

Reply:
Discussed, keeping existing approach